GroundDBMemoryPool/src/rdma/util.cc

```diff
@@ -422,86 +396,10 @@ int resources_create(struct resources *res, const char *serverName, const int tc
         rc = 1;
         goto resources_create_exit;
     }
-    /* each side will send only one WR, so Completion Queue with 1 entry is enough */
-    cq_size = 1;
-    res->cq = ibv_create_cq(res->ib_ctx, cq_size, NULL, NULL, 0);
-    if (!res->cq)
-    {
-        fprintf(stderr, "failed to create CQ with %u entries\n", cq_size);
-        rc = 1;
-        goto resources_create_exit;
-    }
-    /* allocate the memory buffer that will hold the data */
-    size = BUFFER_SIZE;
-    res->buf = (char *)malloc(size);
-    if (!res->buf)
-    {
-        fprintf(stderr, "failed to malloc %Zu bytes to memory buffer\n", size);
-        rc = 1;
-        goto resources_create_exit;
-    }
-    memset(res->buf, 0, size);
-    /* only in the server side put the message in the memory buffer */
-    if (!serverName)
-    {
-        strcpy(res->buf, VERIFIER);
-        fprintf(stdout, "going to send the message: '%s'\n", res->buf);
-    }
-    else
-        memset(res->buf, 0, size);
-    /* register the memory buffer */
-    mr_flags = IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_READ | IBV_ACCESS_REMOTE_WRITE;
-    res->mr = ibv_reg_mr(res->pd, res->buf, size, mr_flags);
-    if (!res->mr)
-    {
-        fprintf(stderr, "ibv_reg_mr failed with mr_flags=0x%x\n", mr_flags);
-        rc = 1;
-        goto resources_create_exit;
-    }
-    fprintf(stdout, "MR was registered with addr=%p, lkey=0x%x, rkey=0x%x, flags=0x%x\n",
-            res->buf, res->mr->lkey, res->mr->rkey, mr_flags);
-    /* create the Queue Pair */
-    memset(&qp_init_attr, 0, sizeof(qp_init_attr));
-    qp_init_attr.qp_type = IBV_QPT_RC;
-    qp_init_attr.sq_sig_all = 1;
-    qp_init_attr.send_cq = res->cq;
-    qp_init_attr.recv_cq = res->cq;
-    qp_init_attr.cap.max_send_wr = 1;
-    qp_init_attr.cap.max_recv_wr = 1;
-    qp_init_attr.cap.max_send_sge = 1;
-    qp_init_attr.cap.max_recv_sge = 1;
-    res->qp = ibv_create_qp(res->pd, &qp_init_attr);
-    if (!res->qp)
-    {
-        fprintf(stderr, "failed to create QP\n");
-        rc = 1;
-        goto resources_create_exit;
-    }
-    fprintf(stdout, "QP was created, QP number=0x%x\n", res->qp->qp_num);
 resources_create_exit:
     if (rc)
     {
         /* Error encountered, cleanup */
-        if (res->qp)
-        {
-            ibv_destroy_qp(res->qp);
-            res->qp = NULL;
-        }
-        if (res->mr)
-        {
-            ibv_dereg_mr(res->mr);
-            res->mr = NULL;
-        }
```

Comment:
Maybe still be necessary for cleanup.

Reply:
In this commit, when calling `resources_create()`, there is not `qp` or `mr` as they will be later created in `connect_qp()` and `register_mr()` respectively